### PR TITLE
F1762 iac all blocks

### PR DESCRIPTION
**Pull request by ssickles**
This PR makes a number of upgrades all aimed at providing the full set of configuration and overrides options for all blocks.

For overrides:
- The ApplicationOverrides, DatastoreOverrides, and SubdomainOverrides have been split out into their own files.
- We have added a new BlockOverrides, NetworkOverrides, ClusterOverrides, ClusterNamespaceOverrides, IngressOverrides, and DomainOverrides structs in order to support these in the overrides.
- Added `Blocks`, `Networks`, `Clusters`, `ClusterNamespaces`, `Ingresses`, and `Domains` to the list of overrides that we now support on a ConfigurationOverrides.
- Removed `Connections` from SubdomainOverrides and DatastoreOverrides. These should have never been here as it doesn't make sense to provide overrides for core structure.

For configurations:
- Added a BlockConfiguration, NetworkConfiguration, ClusterConfiguration, ClusterNamespaceConfiguration, IngressConfiguration, and DomainConfiguration structs in order to support these in the configuration.
- Added `Connections` to AppConfiguration. It was missing and this adds the proper support.
- Added `Blocks`, `Networks`, `Clusters`, `ClusterNamespaces`, `Ingresses`, and `Domains` to the list of configurations we now support in EnvConfigurations.

### CODE REVIEW

File: config/app_configuration.go [hunk 1]
```diff
@@ -9,12 +9,14 @@ import (
...
-	ModuleSource        string                   `yaml:"module" json:"module"`
-	ModuleSourceVersion *string                  `yaml:"module_version,omitempty" json:"moduleVersion"`
-	Variables           map[string]any           `yaml:"vars" json:"vars"`
-	EnvVariables        map[string]string        `yaml:"environment" json:"envVars"`
-	Capabilities        CapabilityConfigurations `yaml:"capabilities" json:"capabilities"`
+	Name                string                 `yaml:"-" json:"name"`
+	ModuleSource        string                 `yaml:"module" json:"module"`
+	ModuleSourceVersion *string                `yaml:"module_version,omitempty" json:"moduleVersion"`
+	Variables           map[string]any         `yaml:"vars" json:"vars"`
+	Connections         core.ConnectionTargets `yaml:"connections" json:"connections"`
```
Comment (BSick7) [inline config/app_configuration.go:16]: One line 70, Connections should be passed into validation, right?
https://github.com/nullstone-io/iac/pull/4/files#diff-295058163fe8dac3bcc2360e6b0867465e79e90494b74819136250c7cb34fc6bL64

File: config/cluster_namespace_configuration.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	Name                string                 `yaml:"-" json:"name"`
+	ModuleSource        string                 `yaml:"module" json:"module"`
+	ModuleSourceVersion *string                `yaml:"module_version,omitempty" json:"moduleVersion"`
+	Variables           map[string]any         `yaml:"vars" json:"vars"`
+	Connections         core.ConnectionTargets `yaml:"connections" json:"connections"`
+}
+
+func (cn ClusterNamespaceConfiguration) Validate(resolver *find.ResourceResolver, configBlocks []core.BlockConfiguration) (errors.ValidationErrors, error) {
+	yamlPath := fmt.Sprintf("cluster_namespaces.%s", cn.Name)
+	return ValidateBlock(resolver, configBlocks, yamlPath, "*/*/*", cn.ModuleSource, *cn.ModuleSourceVersion, cn.Variables, cn.Connections, nil)
```
Comment (BSick7) [inline config/cluster_namespace_configuration.go:20]:
```suggestion
	return ValidateBlock(resolver, configBlocks, yamlPath, "cluster-namespace/*/*", cn.ModuleSource, *cn.ModuleSourceVersion, cn.Variables, cn.Connections, nil)
```

**Review by BSick7 — APPROVED** [review]
Made a couple of notes, but otherwise looks good!

File: config/app_configuration.go (hunk 1 above)
Comment (ssickles) [reply config/app_configuration.go:16]: I think you are right...
From the diffs it looks like this was missing before this PR. I have updated this to pass the connections are you pointed out. Good catch.

File: config/cluster_namespace_configuration.go (hunk 2 above)
Comment (ssickles) [reply config/cluster_namespace_configuration.go:20]: Wups, good catch!

**Pull request merged**